packages/snage/src/command/set.ts

```diff
@@ -0,0 +1,131 @@
+            .example('$0', 'set --on "issue = #22"   issue   # unset issue field')
+            .describe('on', 'Condition for setting values')
+            .positional('field', {type: 'string', describe: 'The field name'})
+            .positional('value', {array: true, type: 'string', describe: 'The field values (Empty if unset)'}) as any,
+    handler: ({config: configFile, field: fieldName, value: stringValue, on}) => {
+        pipe(
+            loadConfig(configFile),
+            E.chain((config) =>
+                pipe(
+                    parseNotes(config, resolveChangelogDirectory(config, configFile)),
+                    E.bimap(errorToString, (notes) => [config, notes] as const)
+                )
+            ),
+            E.chain(([config, notes]) => setHandler({fields: config.fields, notes, condition: on, stringValue, fieldName})),
+            E.fold(printAndExit, (files) => {
+                const err = files.reduce((hasError, f) => {
+                    try {
+                        fs.writeFileSync(f.file, f.content);
+                        console.log(`${f.file}`);
+                        return hasError;
+                    } catch (e) {
+                        console.error(`Error: ${f.file}: ${e}`);
+                        return true;
+                    }
+                }, false);
+                if (err) {
+                    console.error('Could not write all files :/');
+                    process.exit(1);
+                }
+            })
+        );
+    },
+};
+
+const validateFieldAndValue = (fields: Field[], fieldName: string, stringValue: string[]): E.Either<string, unknown> => {
+    const field = fields.find((field) => field.name === fieldName);
+    if (!field) {
+        return E.left(`Field ${fieldName} does not exist`);
+    }
+
+    if (!field.optional && stringValue.length === 0) {
+        return E.left(`${field.name} is required but no values were provided.`);
+    }
+    if (field.optional && stringValue.length === 0) {
+        return E.right(undefined);
+    }
+    if (!field?.list && stringValue.length > 1) {
+        return E.left(`Field ${fieldName} is not a list field. You may only provide one value.`);
+    }
+    const value = field?.list ? stringValue : stringValue[0];
+    return pipe(
+        parseField(field, {[fieldName]: value}, false),
+        E.bimap(
+            (err) => `${value} is not valid: ${JSON.stringify(err)}`,
+            (parsed) => {
+                return parsed[fieldName];
+            }
+        )
+    );
+};
+
+export const setHandler = ({fields, condition, fieldName, stringValue, notes}: Options): E.Either<string, Array<{file: string; content: string}>> => {
+    return pipe(
+        validateFieldAndValue(fields, fieldName, stringValue),
+        E.chain((value) => {
+            const parser = createParser(fields);
+            return pipe(
+                !!condition && condition !== '',
```

Comment:
Da geht aber jemand komplett fp-ts-steil.

Spätestens hier bin ich mir ziemlich sicher, dass man das Filtern der Notes (inklusive boolean-Check) in eine separate Methode packen sollte.
Ein klassisches `if` würde es dann vermutlich auch tun.